### DISCUSSION TITLE
Teenant email address

### DISCUSSION THREAD

**Discussion by tuhin24** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#68] [Q&A] [answered]
How do I access logged in tenant's email address from the lambda functions? (GetProducts(), GetOrders())

**Comment by ujwalbukka** [comment]
@tuhin24 - The email address will be part of the JWT token, you can getting from JWT token. 

In this reference solution, the [tenant authorizer](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/server/Resources/tenant_authorizer.py#L71) already gets other attributes from JWT token, similarly you can add code to get "email" from the JWT token.

**Reply by tuhin24** [reply]
@ujwalbukka , I made changes and deleted the browser cache but I still do not see tenantEmail in the event object in the lambda functions. Does the codepipeline automatically deploy changes made in shared_service_authorizer.py and tenant_authorizer.py? Do I need to manually deploy it?

**Reply by ujwalbukka** [reply, marked as answer]
@tuhin24 - In the reference solution, Codepipeline only deploys application services i.e., it deploys [tenant_template.yaml](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/server/tenant-template.yaml). 
The shared services and the authorizers you mentioned above which are part of [boostrap-template.yaml](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/server/bootstrap-template.yaml) needs to be deployed manually. You can deploy this by just running [deployment.sh](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/deployment.sh) or by running line54 and line55 in the deployment.sh script.
